src/Parsers/IAST.h

```diff
@@ -240,13 +222,117 @@ class IAST : public std::enable_shared_from_this<IAST>, public TypePromotion<IAS
+
+            Hiliter(const Hiliter & other) = delete;
+            Hiliter(Hiliter && other) = delete;
+            Hiliter & operator=(const Hiliter & other) = delete;
+            Hiliter & operator=(Hiliter && other) = delete;
+        };
+        Hiliter createHiliter(const char * hilite_type) const;
+        void writePossiblyHilited(std::string_view str, const char * hilite_type) const;
+        void writeIdentifierOrAlias(const String & name, bool hilite_as_identifier) const;
+
+        FormattingBuffer(WriteBuffer & ostr_, const FormatSettings & settings_, FormatState & state_, FormatStateStacked stacked_state_) :
+            ostr(ostr_), settings(settings_), state(state_), stacked_state(stacked_state_) {}
+    public:
+        FormattingBuffer(WriteBuffer & ostr_, const FormatSettings & settings_, FormatState & state_) :
+            ostr(ostr_), settings(settings_), state(state_) {}
+
+        FormattingBuffer copy() const;
+
+        const FormatSettings & getSettings() const;
+
+        FormattingBuffer copyWithNewSettings(const FormatSettings & settings_) const;
+
+        FormattingBuffer & setIndent(UInt8 indent);
+
+        FormattingBuffer & increaseIndent(int extra_indent = 1);
+
+        FormattingBuffer & setNeedParens(bool value = true);
+
+        FormattingBuffer & setExpressionListAlwaysStartsOnNewLine(bool value = true);
+
+        FormattingBuffer & setExpressionListPrependWhitespace(bool value = true);
+
+        FormattingBuffer & setSurroundEachListElementWithParens(bool value = true);
+
+        FormattingBuffer & setCurrentSelect(const IAST * select);
+
+        bool insertAlias(std::string alias, Hash printed_content) const;
+
+        void writeKeyword(std::string_view str) const;
+        void writeFunction(std::string_view str) const;
+        void writeOperator(std::string_view str) const;
+        void writeSubstitution(std::string_view str) const;
+
+        /** A special hack. If it's [I]LIKE or NOT [I]LIKE expression and the right hand side is a string literal,
+          *  we will highlight unescaped metacharacters % and _ in string literal for convenience.
+          * Motivation: most people are unaware that _ is a metacharacter and forgot to properly escape it with two backslashes.
+          * With highlighting we make it clearly obvious.
+          *
+          * Another case is regexp match. Suppose the user types match(URL, 'www.clickhouse.com'). It often means that the user is unaware
+          *  that . is a metacharacter.
+          */
+        void writeStringLiteralWithMetacharacters(const String & str, const char * metacharacters) const;
+
+        void writeIdentifier(const String & name) const;
+        void writeAlias(const String & name) const;
+        void writeProbablyBackQuotedIdentifier(const String & name) const;
+
+        void nlOrWs() const;  // Write newline or whitespace.
+        void nlOrNothing() const;
+
+        void writeSecret(const String & secret = "") const;
```

Comment:
Let's use `std::string_view` for such arguments always in this class where it doesn't cause troubles.